File: 1-intro/3-smooth-sailing/cpp/commonCharacterCount.cpp
int solution(string s1, string s2) {
    int count1[26] = {0};
    int count2[26] = {0};
    
    for (char c: s1) count1[c - 'a']++;
    for (char c: s2) count2[c - 'a']++;
    
    int r = 0;

    for (int i = 0; i < 26; i++)
        r += min(count1[i], count2[i]);
    
    return r;
}

File: 1-intro/3-smooth-sailing/js/commonCharacterCount.js
const solution = (s1, s2) => {
    let r = 0;
    
    for (let i = 0; i < s1.length; i++) {
        if (s2.includes(s1[i])) {
            s2 = s2.replace(s1[i], '');
            r++;
        }
    }
   
    return r;
}
